### PR TITLE
implement maxItems and see more at sku selector

### PR DESCRIPTION
**Pull request by jgfidelis**
https://fidelis--storecomponents.myvtex.com/classic-shoes/p

WITH MOCKED MULTIPLIED OPTIONS:
https://fidelis2--storecomponents.myvtex.com/classic-shoes/p

Não temos exemplo com muitos SKU para testar, mas é bom ver que não quebra o que ja funciona.

### CODE REVIEW

File: react/components/SKUSelector/components/Variation.js [hunk 1]
```diff
@@ -8,8 +10,21 @@ import { variationShape } from '../utils/proptypes'
 import styles from '../styles.css'
 import { imageUrlForSize, VARIATION_IMG_SIZE } from '../../module/images'
 
-const Variation = ({ variation, onSelectItem, maxSkuPrice, isSelected }) => {
+const THRESHOLD = 2
```
Comment (lbebber) [inline react/components/SKUSelector/components/Variation.js]: "threshold" for what? the name could be clearer I think

File: react/components/SKUSelector/components/Variation.js [hunk 2]
```diff
@@ -8,8 +10,21 @@ import { variationShape } from '../utils/proptypes'
...
+
+const Variation = ({ variation, onSelectItem, maxSkuPrice, isSelected, seeMoreLabel, maxItems }) => {
   const displayImage = isColor(variation.name)
+  const { options } = variation
+  const [showAll, setShowAll] = useState(false)
+  const maxItemsNoThreshold = maxItems - THRESHOLD
```
Comment (lbebber) [inline react/components/SKUSelector/components/Variation.js]: I'm not sure what this means

File: react/components/SKUSelector/components/Variation.js [hunk 3]
```diff
@@ -54,6 +69,13 @@ const Variation = ({ variation, onSelectItem, maxSkuPrice, isSelected }) => {
               </SelectorItem>
             )
           })}
+          {!showAll && shouldCollapse && (
+            <div className={styles.seeMoreButton}>
+              <Button variation="tertiary" onClick={showAllAction} size="small" collapseLeft>
+                <IOMessage id={seeMoreLabel} values={{ quantity: overflowQuantity }} testId={'seeMoreLabel'} />
```
Comment (klzns) [inline react/components/SKUSelector/components/Variation.js]: Nice idea!

Use `data-testid` to have a similar API as react testing library https://testing-library.com/docs/dom-testing-library/api-queries#bytestid
```suggestion
                <IOMessage id={seeMoreLabel} values={{ quantity: overflowQuantity }} data-testId="seeMoreLabel" />
```

File: react/components/SKUSelector/components/SKUSelector.js [hunk 4]
```diff
@@ -23,16 +25,20 @@ const SKUSelector = ({
...
         maxSkuPrice={maxSkuPrice}
+        seeMoreLabel={seeMoreLabel}
+        maxItems={maxItems}
       />
       {shouldShowSecondary && (
         <Variation
           variation={secondaryVariation}
-          onSelectItem={skuId => onSelectSKU(false, skuId)}
-          isSelected={sku => sku.itemId === secondaryVariation.value}
+          onSelectItem={useCallback(skuId => onSelectSKU(false, skuId), [onSelectSKU])}
```
Comment (klzns) [inline react/components/SKUSelector/components/SKUSelector.js]: Hooks should be on the root

See https://reactjs.org/docs/hooks-rules.html

**Pull request merged**